### DISCUSSION TITLE
Suggestion for improving default size (number of results)

### DISCUSSION THREAD

**Discussion by marknuyens** [spatie/laravel-json-api-paginate#49] [Ideas]
Hello there, I've been using your package in my projects and found it kind of redundant to have to specify the default maximum number of pagination results in the configuration of your package, since this configuration may already be present on the Eloquent model itself.

I've written some pseudocode below to demonstrate what the desired outcome could look like. I suppose what I'm proposing is to accept a `NULL` value for the configured `default_size`, in which case it will fall-back to the model's original `perPage`.

I think this may help prevent code duplication by relying on Laravel's own Eloquent model. Anyway, before making any PRs I thought I'd check in first to hear what you have to say.

Anyway, hope this helps.

![image](https://user-images.githubusercontent.com/8502792/109997478-876dc980-7d10-11eb-8d8c-647522f97dcb.png)

**Comment by AlexVanderbist** [comment]
I'd go even further and say that if a model's `perPage` is set, we always use that and otherwise fall back to the default in the config. Feel free to send a PR for this!
